File: C/retangulo.c
#include <stdio.h>

#include <math.h>
int main()
{

  double base, altura, area, perimetro, diagonal;

  printf("Digite a base do retangulo: ");
  scanf("%lf", &base);
  printf("Digite a altura do retangulo: ");
  scanf("%lf", &altura);

  area = base * altura;
  perimetro = 2 * (base + altura);
  diagonal = sqrt(pow(base, 2) + pow(altura, 2));

  printf("Area: %.2lf\n", area);
  printf("Perimetro: %.2lf\n", perimetro);
  printf("Diagonal: %.2lf\n", diagonal);

  return 0;
}

File: C#/Retangulo.cs
using System;
using System.Globalization;

namespace RetanguloApp
{
    public class Retangulo
    {
        public void Calcular()
        {
            CultureInfo CI = CultureInfo.InvariantCulture;

            Console.WriteLine("Base do retangulo: ");
            double largura = double.Parse(Console.ReadLine(), CI);
            Console.WriteLine("Altura do retangulo: ");
            double altura = double.Parse(Console.ReadLine(), CI);

            double area = largura * altura;
            double perimetro = 2 * (largura + altura);
            double diagonal = Math.Sqrt(Math.Pow(largura, 2.0) + Math.Pow(altura, 2.0));

            Console.WriteLine("AREA = " + area.ToString("F4", CI));
            Console.WriteLine("PERIMETRO = " + perimetro.ToString("F4", CI));
            Console.WriteLine("DIAGONAL = " + diagonal.ToString("F4", CI));
        }
    }
}